### PR TITLE
Include developer guide

### PR DESCRIPTION
**Pull request by jschueller**
From the old Latex documentation, mainly updated for git and mingw changes,
refreshed some screenshots.

### CODE REVIEW

**Comment by sofianehaddad** [comment]
+1

**Comment by jschueller** [comment]
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/library_development.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/architecture.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/git_workflow.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/module_development.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/windows_native_port.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/windows_port.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/wrapper_development.rst
https://github.com/jschueller/openturns/blob/developer-guide/python/doc/developer_guide/coding_rules.rst

File: python/doc/developer_guide/library_development.rst [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+If your class introduces important mathematical concepts or impacts the
+library architecture it may be useful to add some more details in the
+latex documentation.
+
+#. Build and download the documentation:
+
+   ::
+
+       svn co https://svn.openturns.org/openturns-doc/trunk
```
Comment (dbarbier) [inline python/doc/developer_guide/library_development.rst]: github here ?

**Comment by dbarbier** [comment]
I do not remember, did we agree on moving developer's guide here?  If yes, we should merge this PR and fix glitches afterwards.

**Pull request merged**